DEVELOPER_GUIDE.md

```diff
@@ -1,43 +1,70 @@
 - [Developer Guide](#developer-guide)
-  - [Linter and Test Suite](#linter-and-test-suite)
-  - [Integration Tests](#integration-tests)
-  - [Build the Documentation with Sphinx](#build-the-documentation-with-sphinx)
+  - [Prerequisites](#prerequisites)
+  - [Running Tests](#running-tests)
+  - [Linter](#linter)
+  - [Documentation](#documentation)
 
 # Developer Guide
 
-## Linter and Test Suite 
-Run the linter and test suite to ensure your changes do not break existing code:
+## Prerequisites
+
+Python 3.6 or newer is required.
+
+```
+$ python --version
+Python 3.11.1
+```
+
+Install [Nox](https://nox.thea.codes/en/stable/) for task management.
 
 ```
-# Install Nox for task management
 $ python -m pip install nox
+```
 
-# Auto-format and lint your changes
-$ nox -rs format
+## Running Tests
+
+Tests require a live instance of OpenSearch running in docker.
+
+This will start a new instance and run tests.
+
+```
+./.ci/run-tests
+```
+
+If your OpenSearch docker instance is running, you can execute the test suite directly.
 
-# Run the test suite
+```
 $ nox -rs test
 ```
 
-## Integration Tests
-To run the integration tests locally, run:
+To run tests against different versions of OpenSearch, use `run-tests [with/without security] [version]`:
```

Comment:
Can we also add that tests can be run against unreleased OpenSearch as well using `./.ci/run-tests opensearch false SNAPSHOT`. To do so: OpenSearch instance can be started by building it from the OpenSearch repo branch. Related workflow: https://github.com/opensearch-project/opensearch-py/blob/main/.github/workflows/integration-unreleased.yml

Reply:
Done.